src/random.cpp

```diff
@@ -65,6 +65,34 @@ static inline int64_t GetPerformanceCounter()
 #endif
 }
 
+#if !defined(_MSC_VER) && (defined(__x86_64__) || defined(__amd64__))
+static bool RDRandSupported()
```

Comment:
I think it would be useful to have a generic cpuid util function for this, for easier support for (among other things) sha extensions in the future.

We can always break it out later, though.